wa/output_processors/postgresql.py

```diff
@@ -0,0 +1,601 @@
+                    metric.units,
+                    metric.lower_is_better))
+            for classifier in metric.classifiers:
+                self.current_classifier_uuid = uuid.uuid4()
+                self.cursor.execute(
+                    self.sql_command['create_classifier'],
+                    (
+                        self.current_classifier_uuid,
+                        None,
+                        self.current_metric_uuid,
+                        classifier,
+                        metric.classifiers[classifier]))
+        # Job augmentations
+        for augmentation in output.spec.augmentations:
+            if not augmentation.startswith('~'):
+                self.current_augmentation_uuid = self.augmentations_already_added[augmentation]
+                self.current_job_aug_uuid = uuid.uuid4()
+                self.cursor.execute(
+                    self.sql_command['create_job_aug'],
+                    (
+                        self.current_job_aug_uuid,
+                        self.current_job_uuid,
+                        self.current_augmentation_uuid))
+            else:
+                continue
+        # Workload parameters (job specific)
+        self.insert_parameters(
+            "workload",
+            output.spec.workload_parameters,
+            job_specific=True)
+        # Runtime parameters (job specific)
+        self.insert_parameters(
+            "runtime",
+            output.spec.runtime_parameters,
+            job_specific=True)
+        self.conn.commit()
+
+    def export_run_output(self, output, target_info):  # pylint: disable=unused-argument, too-many-locals
+        # Update the job statuses following completion of the run
+        for job in output.jobs:
+            job_id = job.id
+            job_status = job.status
+            self.cursor.execute(
+                "UPDATE Jobs SET status=%s WHERE job_id=%s and run_oid=%s",
+                (
+                    job_status,
+                    job_id,
+                    self.run_uuid))
+
+        run_uuid = self.run_uuid
+        # Update the run entry after jobs have completed
+        sql_command_update_run = self.sql_command['update_run']
+        retry_on_status = postgres_convert.ListOfLevel(output.run_config.retry_on_status)
+        self.cursor.execute(
+            sql_command_update_run,
+            (
+                output.event_summary,
+                output.basepath,
+                output.status,
+                output.state.timestamp,
+                output.info.run_name,
+                output.info.project,
+                retry_on_status,
+                output.run_config.max_retries,
+                output.run_config.bail_on_init_failure,
+                output.run_config.allow_phone_home,
+                output.info.uuid,
+                output.info.start_time,
+                output.info.end_time,
+                output.metadata,
+                run_uuid))
```

Comment:
A lot of these should be added during `intialize` -- data should be uploaded as soon as possible.

Reply:
I have made sure to add all of the pieces of data as soon as they become available. Many of these are initially added in the export_job stages, but updated in the export_run stage.

Reply:
Again, there i s no need to write info and config stuff here -- only update things that (may) have changed.